src/ostorlab/serve_app/export_utils.py

```diff
@@ -0,0 +1,226 @@
+
+import io
+import ipaddress
+import json
+import pathlib
+import zipfile
+
+from ostorlab.runtimes.local.models import models
+from ostorlab.serve_app import common
+
+SCAN_JSON = "scan.json"
+ASSET_JSON = "asset.json"
+VULNERABILITY_JSON = "vulnerability.json"
+MOBILE_APP = "mobile.app"
+
+RISK_RATINGS_ORDER = {
+    common.RiskRatingEnum.CRITICAL.name: 8,
+    common.RiskRatingEnum.HIGH.name: 7,
+    common.RiskRatingEnum.MEDIUM.name: 6,
+    common.RiskRatingEnum.LOW.name: 5,
+    common.RiskRatingEnum.POTENTIALLY.name: 4,
+    common.RiskRatingEnum.HARDENING.name: 3,
+    common.RiskRatingEnum.SECURE.name: 2,
+    common.RiskRatingEnum.IMPORTANT.name: 1,
+    common.RiskRatingEnum.INFO.name: 0,
+}
+
+
+def export_scan(scan: models.Scan, export_ide: bool = False) -> bytes:
+    """Export the scan details from the given file data.
+
+    Args:
+        scan: The scan object.
+        export_ide: Whether to export the IDE or not.
+    """
+    fd = io.BytesIO()
+    archive = zipfile.ZipFile(fd, "a", zipfile.ZIP_DEFLATED, True)
+
+    _export_asset(scan.id, archive)
+    _export_scan(scan, archive)
+    _export_vulnz(scan.id, archive)
+
+    if export_ide is True:
+        raise NotImplementedError()
+
+    archive.close()
+    fd.seek(0)
+    return fd.read()
+
+
+def _export_asset(scan_id: int, archive: zipfile.ZipFile) -> None:
+    """Export the asset details to the given archive.
+
+    Args:
+        scan_id: The scan id.
+        archive: The archive object.
+    """
+
+    with models.Database() as session:
+        assets = (
+            session.query(models.Asset).filter(models.Asset.scan_id == scan_id).all()
+        )
+        if len(assets) == 0:
+            return None
+        assets_data = []
+        for asset in assets:
+            asset_type = (
+                asset.type.replace("_", " ").lower()
+                if "_" in asset.type
+                else asset.type
+            )
+            asset_dict = {
+                "tags": None,
+                "type": asset_type.lower(),
+            }
+            if asset.type == "android_file" or asset.type == "ios_file":
+                if asset.path is not None:
+                    file_asset_path = pathlib.Path(asset.path)
+                    if file_asset_path.exists() is False:
+                        raise ValueError(
+                            f"{asset_type.capitalize()} File {asset.path} not found."
+                        )
+                    try:
+                        archive.writestr(MOBILE_APP, file_asset_path.read_bytes())
+                    except Exception:
+                        pass
+
+            elif asset.type == "android_store":
+                asset_dict["package_name"] = asset.package_name
+                asset_dict["application_name"] = asset.application_name
+            elif asset.type == "ios_store":
+                asset_dict["package_name"] = asset.bundle_id
```

Comment:
shouldn't the key be `bundle_id` ? or it doesn't really matter?

Reply:
in RE model we have it as package_name.
![image](https://github.com/Ostorlab/oxo/assets/144013278/8dfcc7ee-f9ff-4497-bdb7-5a4822d2c418)